### ISSUE TITLE
Load failed and stopped with "Too many in flight hints" and IOException

### ISSUE THREAD

**Issue by juliayakovlev** [scylladb/scylla#6438]
*Installation details*
Scylla version (or git commit hash): 4.0.0-0.20200505.d95aa77b62 with build-id c9cf14d33523aca01a1ba502a91eb9d400de9c34
Cluster size: 4 nodes (i3.2xlarge)
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-0ad095d521c760c3f

Test: MV backpressure

Schema: 1 base table and 2 materialized views
```
table_definition: |

  CREATE TABLE IF NOT EXISTS users (
    userid bigint,
    initials int,
    first_name text,
    last_name text,
    password text,
    email text,
    address text,
    userdata blob,
    last_access timeuuid,
    PRIMARY KEY(userid)
  );

extra_definitions:
  - CREATE MATERIALIZED VIEW mview.users_by_address AS SELECT userid, address FROM mview.users WHERE address IS NOT NULL and userid IS NOT NULL PRIMARY KEY (address, userid);
  - CREATE MATERIALIZED VIEW mview.users_by_initials AS SELECT userid, initials FROM mview.users WHERE initials IS NOT NULL and userid IS NOT NULL PRIMARY KEY (initials, userid);
```

Cassandra-stress commands running in parallel with nemesises:
```
"cassandra-stress user profile=/tmp/c-s_profile_2mv_2queries.yaml ops'(insert=1)' cl=QUORUM duration=5760m -port jmx=6868 -mode cql3 native -pop seq=250000000..500000000 -rate threads=200"

 "cassandra-stress user profile=/tmp/c-s_profile_2mv_2queries.yaml ops'(mv_p_read1=1,mv_p_read2=1)' cl=QUORUM duration=5760m -port jmx=6868 -mode cql3 native -pop seq=1..250000000 -rate threads=10"
```

List of nemesises before load failure:
```
< t:2020-05-11 23:51:25,648 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method no_corrupt_repair
< t:2020-05-12 00:48:52,350 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method show_toppartitions
< t:2020-05-12 00:59:53,459 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method toggle_table_ics
< t:2020-05-12 01:03:17,296 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method modify_table
< t:2020-05-12 01:11:22,445 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method nodetool_enospc
< t:2020-05-12 01:22:16,477 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method nodetool_refresh
< t:2020-05-12 01:32:26,355 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method repair_streaming_err
< t:2020-05-12 02:14:22,206 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method abort_repair
< t:2020-05-12 07:36:54,898 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method show_toppartitions
< t:2020-05-12 07:47:08,162 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method stop_wait_start_scylla_server
< t:2020-05-12 08:00:45,597 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method major_compaction
< t:2020-05-12 08:16:46,992 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method delete_10_full_partitions
< t:2020-05-12 08:19:22,786 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method nodetool_enospc
< t:2020-05-12 08:30:52,646 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method toggle_table_ics
< t:2020-05-12 08:34:14,353 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method show_toppartitions
< t:2020-05-12 08:44:25,880 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method destroy_data_then_repair
< t:2020-05-12 11:53:14,487 f:nemesis.py      l:657  c:sdcm.nemesis         p:INFO  > sdcm.nemesis.ChaosMonkey: >>>>>>>>>>>>>Started random_disrupt_method abort_repair
```

Load failed during destroy_data_then_repair nemesis running on node 4 (10.0.250.135):
- stop node
- destroy few DB files
- start node and run repair

There are a lot of ```Too many in flight hints``` messages in the cluster:
```
< t:2020-05-12 08:45:17,995 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: WARN  08:45:17,974 Host ip-10-0-18-240.eu-west-1.compute.internal/10.0.18.240:9042 is overloaded.
< t:2020-05-12 08:45:18,000 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: WARN  08:45:17,974 Host ip-10-0-18-240.eu-west-1.compute.internal/10.0.18.240:9042 is overloaded.
< t:2020-05-12 08:45:18,015 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: com.datastax.driver.core.exceptions.OverloadedException: Queried host (ip-10-0-18-240.eu-west-1.compute.internal/10.0.18.240:9042) was overloaded: Too many in flight hints: 10495942
< t:2020-05-12 08:45:18,078 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: com.datastax.driver.core.exceptions.OverloadedException: Queried host (ip-10-0-18-240.eu-west-1.compute.internal/10.0.18.240:9042) was overloaded: Too many in flight hints: 10495942
< t:2020-05-12 08:45:18,084 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: WARN  08:45:17,976 Host ip-10-0-18-240.eu-west-1.compute.internal/10.0.18.240:9042 is overloaded.
< t:2020-05-12 08:45:18,109 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: WARN  08:45:17,976 Host ip-10-0-18-240.eu-west-1.compute.internal/10.0.18.240:9042 is overloaded.
```

and ```OperationTimedOutException``` :
```
< t:2020-05-12 08:47:55,670 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: com.datastax.driver.core.exceptions.OperationTimedOutException: [ip-10-0-250-135.eu-west-1.compute.internal/10.0.250.135:9042] Timed out waiting for server response
< t:2020-05-12 08:47:55,674 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: com.datastax.driver.core.exceptions.OperationTimedOutException: [ip-10-0-250-135.eu-west-1.compute.internal/10.0.250.135:9042] Timed out waiting for server response
```

IOException from cassandra-stress:
```
< t:2020-05-12 08:47:55,670 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: com.datastax.driver.core.exceptions.OperationTimedOutException: [ip-10-0-250-135.eu-west-1.compute.internal/10.0.250.135:9042] Timed out waiting for server response
< t:2020-05-12 08:47:55,674 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@10.0.231.144]: com.datastax.driver.core.exceptions.OperationTimedOutException: [ip-10-0-250-135.eu-west-1.compute.internal/10.0.250.135:9042] Timed out waiting for server response
```

mutation_write_timeout_exception:
```
< t:2020-05-12 08:49:48,214 f:cluster.py      l:1408 c:sdcm.cluster         p:DEBUG > 2020-05-12T08:49:33+00:00  longevity-2mv-backpressure-4d-mv-ba-db-node-11230414-1 !WARNING | scylla: [shard 0] seastar - Exceptional future ignored: exceptions::mutation_write_timeout_exception (Operation timed out for mview.users_by_address - received only 0 responses from 1 CL=ANY.), 
backtrace:   
0x331b31d#012  
0x2e30f07#012  
0x1d71c84#012  
0x2e7ac30#012  
0x2e7ae3f#012  
0x2eb2df5#012
0x2e189b7#012  
0x2e19aee#012  
0xd015a7#012  
/opt/scylladb/libreloc/libc.so.6+0x271a2#012  
0xc1d02d

backtrace<seastar::current_backtrace()::<lambda(seastar::frame)> > at /usr/include/fmt/format.h:1158
 (inlined by) seastar::current_backtrace() at ./build/release/seastar/./seastar/src/util/backtrace.cc:84
seastar::report_failed_future(std::__exception_ptr::exception_ptr const&) at ./build/release/seastar/./seastar/src/core/future.cc:101
seastar::future_state_base::~future_state_base() at /usr/include/fmt/format.h:1158
 (inlined by) seastar::future_state<>::~future_state() at ././seastar/include/seastar/core/future.hh:427
 (inlined by) seastar::future<>::~future() at ././seastar/include/seastar/core/future.hh:977
 (inlined by) operator() at ././seastar/include/seastar/core/future.hh:1284
 (inlined by) run_and_dispose at ././seastar/include/seastar/core/future.hh:522
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/include/fmt/format.h:1158
seastar::reactor::run_some_tasks() at /usr/include/fmt/format.h:1158
seastar::reactor::run_some_tasks() at /usr/include/fmt/format.h:1158
 (inlined by) seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2693
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at /usr/include/boost/program_options/variables_map.hpp:146
main at ./main.cc:482
?? ??:0
_start at ??:?
?? ??:0
```

Cassandra-stress processes are stuck on the loaders:
```
[centos@longevity-2mv-backpressure-4d-mv-ba-loader-node-11230414-1 ~]$ ps -ef | grep cassandra-stress
centos    4463  2027  0 May11 ?        00:00:00 bash -c echo TAG: loader_idx:0-cpu_idx:0-keyspace_idx:1; STRESS_TEST_MARKER=WVSN3S5H54S70ZXX6BWP; cassandra-stress user profile=/tmp/c-s_profile_2mv_2queries.yaml ops'(insert=1)' cl=QUORUM duration=5760m -port jmx=6868 -mode cql3 native -pop seq=250000000..500000000 -rate threads=200 -node 10.0.59.66
centos    4474  4463  0 May11 ?        00:00:00 /bin/sh /usr/bin/cassandra-stress user profile=/tmp/c-s_profile_2mv_2queries.yaml ops(insert=1) cl=QUORUM duration=5760m -port jmx=6868 -mode cql3 native -pop seq=250000000..500000000 -rate threads=200 -node 10.0.59.66
centos    5004  2027  0 May11 ?        00:00:00 bash -c echo TAG: loader_idx:0-cpu_idx:0-keyspace_idx:1; STRESS_TEST_MARKER=SAG36CS6MMTHPLVAJLTK; cassandra-stress user profile=/tmp/c-s_profile_2mv_2queries.yaml ops'(mv_p_read1=1,mv_p_read2=1)' cl=QUORUM duration=5760m -port jmx=6868 -mode cql3 native -pop seq=1..250000000 -rate threads=10 -node 10.0.59.66
centos    5015  5004  0 May11 ?        00:00:00 /bin/sh /usr/bin/cassandra-stress user profile=/tmp/c-s_profile_2mv_2queries.yaml ops(mv_p_read1=1,mv_p_read2=1) cl=QUORUM duration=5760m -port jmx=6868 -mode cql3 native -pop seq=1..250000000 -rate threads=10 -node 10.0.59.66
```

Graphs from Grafana.

Requests:
![Screenshot from 2020-05-12 18-51-04](https://user-images.githubusercontent.com/34435448/81716735-65a20e00-9482-11ea-88fc-e816ae42fb88.png)

Load per instance:
![Screenshot from 2020-05-12 18-51-17](https://user-images.githubusercontent.com/34435448/81716780-75215700-9482-11ea-96c5-b2752f7f861f.png)

Write timeouts:
![Screenshot from 2020-05-12 18-51-36](https://user-images.githubusercontent.com/34435448/81716845-85d1cd00-9482-11ea-934f-0d29c2dfa251.png)

Test id: ```11230414-4967-4d03-9e2b-c38caee41d74```

**Comment by nyh** [comment]
You said  that at 2020-05-12 08:44:25,880 you stopped a node.
Ad that point, writes start to collect hints, and this perfectly normal.
Then less than a minute later, at 2020-05-12 08:45:17,995, you get: "Too many in flight hints: 10495942"

@psarna can you please remind me what are "in-flight hints"? Is this the number of hints saved on disk (if so, why limit them to only a million), or what?

**Comment by vladzcloudius** [comment]
@juliayakovlev @nyh Too-many-hints-in-flight error is a symptom of the `overloaded_exception` thrown from `create_write_response_handler_helper` constructor which eventually is forwarded all up to the CQL and bumps up `writes_unavaliable` metric: see `storage_proxy::mutate_end()`.

It happens when hints are generated faster than I/O backend is capable to write them to the disk.

As a result writes to the corresponding destination are being failed early. This is a part of our admission control.

**Comment by slivne** [comment]
@haaawk / @piodul - please check this out - please note this was run on 4.0.0 since its assume to be "good enough". 

Unless we are overloading the nodes by the customer write workload (which I doubt is the case) - it seems that repair + hints maybe overloading the node (can you please check that the resources repair+hints make sense. 

@piodul your change moving hints to its own verb (making sure its part of the streaming scheduling group) - is that part of 4.0 ?

**Comment by juliayakovlev** [comment]
Same issue receive in the ```2020.1.rc1-0.20200624.6f2a0191009```version.
EncryptionAtRest/EaR-longevity-kmip-200gb-6h-test
Test id: 74931595-c199-405a-b374-ed19d4e6a752

Cassandra-stress load failed:
```
2020-06-25 20:59:44.342: (CassandraStressEvent Severity.CRITICAL): type=failure node=Node longevity-encrypt-at-rest-200gb-6h--loader-node-74931595-2 [54.197.70.93 | 172.30.0.211] (seed: False)
Stress command completed with bad status 1: Failed to connect over JMX; not collecting these stats
Failed to connect over JMX; not collecting th
2020-06-25 17:43:32.729: (CassandraStressLogEvent Severity.ERROR): type=IOException regex=java\.io\.IOException line_number=5845 node=Node longevity-encrypt-at-rest-200gb-6h--loader-node-74931595-2 [54.197.70.93 | 172.30.0.211] (seed: False)
com.datastax.driver.core.exceptions.OverloadedException: Queried host (ip-172-30-0-102.ec2.internal/172.30.0.102:9042) was overljava.io.IOException: Operation x10 on key(s) [323832304c3939354e30]: Error executing: (OverloadedException): Queried host (ip-172-30-0-102.ec2.internal/172.30.0.102:9042) was overloaded: Too many in flight hints: 10486782
```

First time this error is received during nodetool_drain nemesis:
```
< t:2020-06-25 17:43:28,784 f:remote.py       l:739  c:sdcm.remote          p:INFO  > RemoteCmdRunner [centos@172.30.0.211]: com.datastax.driver.core.exceptions.OverloadedException: Queried host (ip-172-30-0-102.ec2.internal/172.30.0.102:9042) was overloaded: Too many in flight hints: 10486782
```

List of all nemesises:
```
2020-06-25 15:06:36.426: (DisruptionEvent Severity.NORMAL): type=start name=NodetoolCleanupMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-1 [52.91.86.24 | 172.30.0.29] (seed: True) duration=None
2020-06-25 15:13:06.809: (DisruptionEvent Severity.NORMAL): type=start name=NodetoolCleanupMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-2 [3.89.35.185 | 172.30.0.100] (seed: False) duration=None
2020-06-25 15:17:26.838: (DisruptionEvent Severity.NORMAL): type=start name=NodetoolCleanupMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-3 [52.91.227.62 | 172.30.0.102] (seed: False) duration=None
2020-06-25 15:22:47.867: (DisruptionEvent Severity.NORMAL): type=start name=NodetoolCleanupMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-4 [34.238.117.119 | 172.30.0.53] (seed: False) duration=None
2020-06-25 15:38:28.816: (DisruptionEvent Severity.NORMAL): type=start name=Drainer node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-3 [52.91.227.62 | 172.30.0.102] (seed: False) duration=None
2020-06-25 15:49:29.934: (DisruptionEvent Severity.NORMAL): type=start name=TruncateMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-2 [3.89.35.185 | 172.30.0.100] (seed: False) duration=None
2020-06-25 16:03:18.792: (DisruptionEvent Severity.NORMAL): type=start name=StopWaitStartService node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-2 [3.89.35.185 | 172.30.0.100] (seed: False) duration=None
2020-06-25 16:19:09.177: (DisruptionEvent Severity.NORMAL): type=start name=StopStartService node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-3 [52.91.227.62 | 172.30.0.102] (seed: False) duration=None
2020-06-25 16:29:26.872: (DisruptionEvent Severity.NORMAL): type=start name=SoftRebootNode node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-4 [34.238.117.119 | 172.30.0.53] (seed: False) duration=None
2020-06-25 16:42:21.511: (DisruptionEvent Severity.NORMAL): type=start name=NodetoolCleanupMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-1 [52.91.86.24 | 172.30.0.29] (seed: True) duration=None
2020-06-25 16:57:45.637: (DisruptionEvent Severity.NORMAL): type=start name=NodetoolCleanupMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-2 [3.89.35.185 | 172.30.0.100] (seed: False) duration=None
2020-06-25 17:06:23.216: (DisruptionEvent Severity.NORMAL): type=start name=NodetoolCleanupMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-3 [52.91.227.62 | 172.30.0.102] (seed: False) duration=None
2020-06-25 17:15:20.477: (DisruptionEvent Severity.NORMAL): type=start name=NodetoolCleanupMonkey node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-4 [34.238.117.119 | 172.30.0.53] (seed: False) duration=None
2020-06-25 17:33:21.284: (DisruptionEvent Severity.NORMAL): type=start name=ShowTopPartitions node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-2 [3.89.35.185 | 172.30.0.100] (seed: False) duration=None
2020-06-25 17:43:07.972: (DisruptionEvent Severity.NORMAL): type=start name=Drainer node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-2 [3.89.35.185 | 172.30.0.100] (seed: False) duration=None
2020-06-25 17:51:24.954: (DisruptionEvent Severity.NORMAL): type=start name=Decommission node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-4 [34.238.117.119 | 172.30.0.53] (seed: False) duration=None
2020-06-25 19:23:32.697: (DisruptionEvent Severity.NORMAL): type=start name=Decommission node=Node longevity-encrypt-at-rest-200gb-6h--db-node-74931595-2 [3.89.35.185 | 172.30.0.100] (seed: False) duration=None
```

Get logs:
```
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                 Log links for testrun with test id 74931595-c199-405a-b374-ed19d4e6a752                                                                                                  |
+-----------------+-------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Date            | Log type    | Link                                                                                                                                                                                                                                     |
+-----------------+-------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 20190101_010101 | prometheus  | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/prometheus_snapshot_20200625_211803.tar.gz                                                                                                           |
| 20200625_205955 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/20200625_205955/grafana-screenshot-overview-20200625_210340-longevity-encrypt-at-rest-200gb-6h--monitor-node-74931595-1.png                          |
| 20200625_205955 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/20200625_205955/grafana-screenshot-scylla-per-server-metrics-nemesis-20200625_205955-longevity-encrypt-at-rest-200gb-6h--monitor-node-74931595-1.png |
| 20200625_210724 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/20200625_210724/grafana-screenshot-overview-20200625_211125-longevity-encrypt-at-rest-200gb-6h--monitor-node-74931595-1.png                          |
| 20200625_210724 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/20200625_210724/grafana-screenshot-scylla-per-server-metrics-nemesis-20200625_210724-longevity-encrypt-at-rest-200gb-6h--monitor-node-74931595-1.png |
| 20200625_212057 | db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/20200625_212057/db-cluster-74931595.zip                                                                                                              |
| 20200625_212057 | loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/20200625_212057/loader-set-74931595.zip                                                                                                              |
| 20200625_212057 | monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/20200625_212057/monitor-set-74931595.zip                                                                                                             |
| 20200625_212057 | sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/74931595-c199-405a-b374-ed19d4e6a752/20200625_212057/sct-runner-74931595.zip                                                                                                              |
+-----------------+-------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Get monitor:
```
hydra investigate show-monitor 74931595-c199-405a-b374-ed19d4e6a752
```

**Comment by roydahan** [comment]
We're still facing this issue on **4.4.dev-0.20201203.c7311d10806**.

The test scenario is: **longevity-50gb-3days**
It happens during **CleanupNemesis** while running nodetool cleanup.

There are various load types on this test scenario:
- c-s (default) Write
- c-s (default) Mixed
- c-s (default) read
- c-s user profile with MV
- c-s write & read with all types of compressions (LZ4Compressor, ZstdCompressor, SnappyCompressor, DeflateCompressor)

**Comment by amoskong** [comment]
I didn't reproduced this issue with 4.4.dev-0.20201229.700ddd191

longevity-50gb-3days (only running CleanupNemesis), the two jobs were only executed 6 hours (default timeout of byo longevity is 6 hours)

- hinted handoff is enabled by default
  - test id: e959e0d9-805d-4616-80e2-5f5a9b11ebdb
  - https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/byo-longevity-test/56/
- hinted handoff is disabled
  - test id: 5d2ee40a-c50b-49a4-a920-762bae8d0abe
  - https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/byo-test/71/console

**Comment by roydahan** [comment]
Please check which version of the test you used.
In this commit d9634753db27cb74fd334d20c720b7200f603502 I reduced the load on this test in order to avoid the issue.
Please check you didn't use this newer version of the test.

**Comment by amoskong** [comment]
> @amoskong,
> Please check which version of the test you used.
> In this commit d9634753db27cb74fd334d20c720b7200f603502 I reduced the load on this test in order to avoid the issue.
> Please check you didn't use this newer version of the test.

You commit has been merged to any branch, so my jobs weren't effected by it.

link: https://github.com/scylladb/scylla-cluster-tests/commit/d9634753db27cb74fd334d20c720b7200f603502

![image](https://user-images.githubusercontent.com/309708/104541918-851ab780-565d-11eb-9716-efe18b4429ab.png)

**Comment by roydahan** [comment]
My PR was merged into master: https://github.com/scylladb/scylla-cluster-tests/pull/3055.
Check what your test yaml used.

**Comment by amoskong** [comment]
> My PR was merged into master: [scylladb/scylla-cluster-tests#3055](https://github.com/scylladb/scylla-cluster-tests/pull/3055).
> Check what your test yaml used.

OK, your commit (https://github.com/scylladb/scylla-cluster-tests/commit/d9ef3bbe9ae119e01cbfaf8e665c235cf86d044f) exists in my tests. I will retest after reverting the patch.

**Comment by amoskong** [comment]
> > My PR was merged into master: [scylladb/scylla-cluster-tests#3055](https://github.com/scylladb/scylla-cluster-tests/pull/3055).
> > Check what your test yaml used.
> 
> OK, your commit ([scylladb/scylla-cluster-tests@d9ef3bb](https://github.com/scylladb/scylla-cluster-tests/commit/d9ef3bbe9ae119e01cbfaf8e665c235cf86d044f)) exists in my tests. I will retest after reverting the patch.

Update:

After retested with old SCT without Roy's commit (https://github.com/scylladb/scylla-cluster-tests/commit/d9ef3bbe9ae119e01cbfaf8e665c235cf86d044f), The scylla issue (https://github.com/scylladb/scylla/issues/6438) can ONLY be reproduced when hinted handoff is enabled.

- cleanup nemesis job, hinted handoff: disabled, not reproduce
  - https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/byo-test/72/
- cleanup nemesis job, hinted handoff: enabled, reproduced
  - https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/byo-longevity-test/57/

**Comment by amoskong** [comment]
> > > My PR was merged into master: [scylladb/scylla-cluster-tests#3055](https://github.com/scylladb/scylla-cluster-tests/pull/3055).
> > > Check what your test yaml used.
> > 
> > 
> > OK, your commit ([scylladb/scylla-cluster-tests@d9ef3bb](https://github.com/scylladb/scylla-cluster-tests/commit/d9ef3bbe9ae119e01cbfaf8e665c235cf86d044f)) exists in my tests. I will retest after reverting the patch.
> 
> Update:
> 
> After retested with old SCT without Roy's commit ([scylladb/scylla-cluster-tests@d9ef3bb](https://github.com/scylladb/scylla-cluster-tests/commit/d9ef3bbe9ae119e01cbfaf8e665c235cf86d044f)), The scylla issue (#6438) can ONLY be reproduced when hinted handoff is enabled.
> 
> * cleanup nemesis job, hinted handoff: disabled, not reproduce
>   
>   * https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/byo-test/72/
> * cleanup nemesis job, hinted handoff: enabled, reproduced


>   * https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/byo-longevity-test/57/

@roydahan 

In above job (hinted handoff enabled, reproduced),  I only found `WriteTimeout Exception` in `write and mixed` workloads. Other (`read`) workloads `all` finished `Successfully`.

### Conclusion: 
So there is no data lose when https://github.com/scylladb/scylla/issues/6438 is reproduced.

**Comment by roydahan** [comment]
So, if IIUC, we have c-s writes failing on this scenario when HH is enabled.

**Comment by amoskong** [comment]
> So, if IIUC, we have c-s writes failing on this scenario when HH is enabled.

Yes.

**Comment by k0machi** [comment]
Another CDC Longevity with same issue:
```

2022-04-17 04:44:46.116: (CassandraStressLogEvent Severity.ERROR) period_type=one-time event_id=d412920d-9967-456a-9817-410847c1c071: type=TooManyHintsInFlight regex=Too many hints in flight|Too many in flight hints line_number=23800 node=Node longevity-cdc-100gb-4h-master-loader-node-e2c651c6-2 [54.76.149.37 | 10.0.0.31] (seed: False)
com.datastax.driver.core.exceptions.OverloadedException: Queried host (10.0.1.67/10.0.1.67:9042) was overloaded: Too many in flight hints: 10485778

2022-04-17 04:44:46.116: (CassandraStressLogEvent Severity.ERROR) period_type=one-time event_id=d412920d-9967-456a-9817-410847c1c071: type=TooManyHintsInFlight regex=Too many hints in flight|Too many in flight hints line_number=23804 node=Node longevity-cdc-100gb-4h-master-loader-node-e2c651c6-2 [54.76.149.37 | 10.0.0.31] (seed: False)
com.datastax.driver.core.exceptions.OverloadedException: Queried host (10.0.1.67/10.0.1.67:9042) was overloaded: Too many in flight hints: 10485778

2022-04-17 04:44:46.580: (CassandraStressLogEvent Severity.ERROR) period_type=one-time event_id=9db2b353-7f99-448b-ac1e-ffdc53ac938e: type=TooManyHintsInFlight regex=Too many hints in flight|Too many in flight hints line_number=23038 node=Node longevity-cdc-100gb-4h-master-loader-node-e2c651c6-1 [34.243.29.154 | 10.0.2.178] (seed: False)
com.datastax.driver.core.exceptions.OverloadedException: Queried host (10.0.1.67/10.0.1.67:9042) was overloaded: Too many in flight hints: 10485778

2022-04-17 04:44:47.018: (CassandraStressLogEvent Severity.ERROR) period_type=one-time event_id=d412920d-9967-456a-9817-410847c1c071: type=TooManyHintsInFlight regex=Too many hints in flight|Too many in flight hints line_number=23807 node=Node longevity-cdc-100gb-4h-master-loader-node-e2c651c6-2 [54.76.149.37 | 10.0.0.31] (seed: False)
com.datastax.driver.core.exceptions.OverloadedException: Queried host (10.0.1.67/10.0.1.67:9042) was overloaded: Too many in flight hints: 10485778

2022-04-17 04:44:47.019: (CassandraStressLogEvent Severity.ERROR) period_type=one-time event_id=d412920d-9967-456a-9817-410847c1c071: type=TooManyHintsInFlight regex=Too many hints in flight|Too many in flight hints line_number=23813 node=Node longevity-cdc-100gb-4h-master-loader-node-e2c651c6-2 [54.76.149.37 | 10.0.0.31] (seed: False)
com.datastax.driver.core.exceptions.OverloadedException: Queried host (10.0.1.67/10.0.1.67:9042) was overloaded: Too many in flight hints: 10485778

```

## Installation details

Kernel Version: 5.13.0-1021-aws
Scylla version (or git commit hash): `5.1.dev-20220415.75786c42cb2e` with build-id `624bf766a24fba740b118c805912a1b94df59dc5`
Cluster size: 6 nodes (i3.4xlarge)

Scylla Nodes used in this run:
- longevity-cdc-100gb-4h-master-db-node-e2c651c6-8 (3.251.88.249 | 10.0.0.148) (shards: 14)
- longevity-cdc-100gb-4h-master-db-node-e2c651c6-7 (34.242.229.1 | 10.0.3.135) (shards: 14)
- longevity-cdc-100gb-4h-master-db-node-e2c651c6-6 (3.250.144.218 | 10.0.0.149) (shards: 14)
- longevity-cdc-100gb-4h-master-db-node-e2c651c6-5 (34.249.72.236 | 10.0.1.169) (shards: 14)
- longevity-cdc-100gb-4h-master-db-node-e2c651c6-4 (54.171.190.39 | 10.0.2.130) (shards: 14)
- longevity-cdc-100gb-4h-master-db-node-e2c651c6-3 (52.48.1.165 | 10.0.1.214) (shards: 14)
- longevity-cdc-100gb-4h-master-db-node-e2c651c6-2 (3.250.189.245 | 10.0.2.111) (shards: 14)
- longevity-cdc-100gb-4h-master-db-node-e2c651c6-1 (34.253.220.78 | 10.0.1.67) (shards: 14)


OS / Image: `ami-0ac9cc9b9c7b1a0f0` (aws: eu-west-1)

Test: `longevity-cdc-100gb-4h-test`
Test id: `e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f`
Test name: `scylla-master/longevity/longevity-cdc-100gb-4h-test`
Test config file(s):

- [longevity-cdc-100gb-4h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/1db1271a4a1de09f1d87b1f43e367e8cedca2af1/test-cases/longevity/longevity-cdc-100gb-4h.yaml)

## Issue description

**>>>>>>>**
**Your description here...**
**<<<<<<<**

- Restore Monitor Stack command: `$ hydra investigate show-monitor e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f)
- Show all stored logs command: `$ hydra investigate show-logs e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f`

## Logs:
- **db-cluster-e2c651c6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/db-cluster-e2c651c6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/db-cluster-e2c651c6.tar.gz)
- **monitor-set-e2c651c6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/monitor-set-e2c651c6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/monitor-set-e2c651c6.tar.gz)
- **loader-set-e2c651c6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/loader-set-e2c651c6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/loader-set-e2c651c6.tar.gz)
- **normal-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/normal-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/normal-e2c651c6.log.tar.gz)
- **summary-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/summary-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/summary-e2c651c6.log.tar.gz)
- **events-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/events-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/events-e2c651c6.log.tar.gz)
- **output-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/output-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/output-e2c651c6.log.tar.gz)
- **debug-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/debug-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/debug-e2c651c6.log.tar.gz)
- **sct-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/sct-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/sct-e2c651c6.log.tar.gz)
- **error-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/error-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/error-e2c651c6.log.tar.gz)
- **critical-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/critical-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/critical-e2c651c6.log.tar.gz)
- **raw_events-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/raw_events-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/raw_events-e2c651c6.log.tar.gz)
- **warning-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/warning-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/warning-e2c651c6.log.tar.gz)
- **email_data-e2c651c6.json.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/email_data-e2c651c6.json.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/email_data-e2c651c6.json.tar.gz)
- **argus-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/argus-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/argus-e2c651c6.log.tar.gz)
- **left_processes-e2c651c6.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/left_processes-e2c651c6.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/left_processes-e2c651c6.log.tar.gz)
- **parallel-timelines-report-e2c651c6.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/parallel-timelines-report-e2c651c6.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e2c651c6-d496-4bc8-a0b2-c43f3cd1c21f/20220417_074525/parallel-timelines-report-e2c651c6.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/longevity/job/longevity-cdc-100gb-4h-test/327/)

**Comment by DoronArazii** [comment]
Removing "master/triage" - we have @piodul on this issue
